### PR TITLE
feat(react-native): introduce chat in the RN Dogfood app

### PR DESCRIPTION
**Pull request by khushal87**

### CODE REVIEW

File: sample-apps/react-native/dogfood/src/screens/Meeting/ChatScreen.tsx [hunk 1]
```diff
@@ -26,9 +26,13 @@ const ChannelHeader = () => {
     (store) => store.chatLabelNoted,
   );
   const appStoreSetState = useAppGlobalStoreSetState();
-  const [noted, setNoted] = useState<boolean>(!!chatLabelNoted);
+  const [isNoted, setIsNoted] = useState<boolean>(!!chatLabelNoted);
```
Comment (vanGalilea) [inline sample-apps/react-native/dogfood/src/screens/Meeting/ChatScreen.tsx:29]: When var has `boolean` type we use prefix is thus `isNoted`

File: sample-apps/react-native/dogfood/src/screens/Meeting/ChatScreen.tsx [hunk 2]
```diff
@@ -26,9 +26,13 @@ const ChannelHeader = () => {
...
 
-  return !noted ? (
+  if (isNoted) {
```
Comment (vanGalilea) [inline sample-apps/react-native/dogfood/src/screens/Meeting/ChatScreen.tsx:31]: Early exits over ternary ops in inline return, makes it more readable

File: sample-apps/react-native/dogfood/src/screens/Meeting/ChatScreen.tsx [hunk 3]
```diff
@@ -70,30 +74,23 @@ export const ChatScreen = ({ route }: ChatScreenProps) => {
   }
 
   return (
-    <SafeAreaView style={styles.container}>
-      {channel && (
```
Comment (vanGalilea) [inline sample-apps/react-native/dogfood/src/screens/Meeting/ChatScreen.tsx]: Was unneeded due to check in line 72

File: sample-apps/react-native/dogfood/src/components/MeetingUI.tsx [hunk 4]
```diff
@@ -64,7 +68,15 @@ export const MeetingUI = ({
...
           setMode={setSelectedMode}
         />
-        <ActiveCall mode={selectedMode} />
+        <ActiveCall
+          mode={selectedMode}
+          chatButton={{
+            onPressHandler: () => {
+              navigation.navigate('ChatScreen', { callId: callId });
+            },
+            unreadBadgeCountIndicator: unreadCount,
```
Comment (vanGalilea) [inline sample-apps/react-native/dogfood/src/components/MeetingUI.tsx:77]:
```suggestion
            unreadBadgeCountIndicator,
```

File: sample-apps/react-native/dogfood/src/components/MeetingUI.tsx [hunk 5]
```diff
@@ -30,6 +32,8 @@ export const MeetingUI = ({
 }: Props) => {
   const [selectedMode, setSelectedMode] = React.useState<Mode>('grid');
   const call = useCall();
+  const channelWatched = useChannelWatch();
+  const unreadCount = useUnreadCount({ channelWatched });
```
Comment (vanGalilea) [inline sample-apps/react-native/dogfood/src/components/MeetingUI.tsx:36]:
```suggestion
  const unreadBadgeCountIndicator = useUnreadCount({ channelWatched });
```

Comment (khushal87) [reply sample-apps/react-native/dogfood/src/components/MeetingUI.tsx:36]: This hook returns a number, so probably unreadCount made more sense

File: sample-apps/react-native/dogfood/src/screens/Meeting/MeetingScreen.tsx [hunk 6]
```diff
@@ -22,6 +24,7 @@ export const MeetingScreen = (props: Props) => {
 
   const onJoin = () => {
     setShow('active-call');
+    appStoreSetState({ chatLabelNoted: false });
```
Comment (vanGalilea) [inline sample-apps/react-native/dogfood/src/screens/Meeting/MeetingScreen.tsx:27]: What is the idea behind having a global state for this? Local one should be enough as we reset it everytime

**Review by vanGalilea — CHANGES_REQUESTED** [review]

**Review by vanGalilea — APPROVED** [review]
Good job @khushal87 🥳

**Pull request merged**